pkg/cloudmap/resource_manager.go

```diff
@@ -385,3 +390,32 @@ func (m *defaultResourceManager) isCloudMapServiceOwnedByVirtualNode(ctx context
 func (m *defaultResourceManager) buildCloudMapServiceSummaryCacheKey(nsSummary *servicediscovery.NamespaceSummary, serviceName string) string {
 	return fmt.Sprintf("%s/%s", awssdk.StringValue(nsSummary.Id), serviceName)
 }
+
+func (m *defaultResourceManager) getClusterNodeInfo(ctx context.Context) map[string]nodeAttributes {
```

Comment:
Is there a possibility of this going out of sync when a virtualNode is not touched but it's corresponding pod moves to a new node, where the new node is added after this function was called or the reconcile also takes into account Pod updates as well eliminating any possibility of out of sync?

Reply:
Yeah, CloudMap controller watches for pod events as well. So, we should be good with the above scenario..